test/distributed/tensor/parallel/test_tp_random_state.py

```diff
@@ -0,0 +1,185 @@
+                assert _1d_mesh.ndim == 1
+
+                tensor_local = dtensor.to_local()
+                local_shape = tensor_local.shape
+
+                # all-gather local shards
+                tensor_gather = _1d_mesh.all_gather(tensor_local, gather_dim=0)
+
+                # compare local shards within the TP group
+                self.assertEqual(_1d_mesh.get_coordinate()[0], tp_rank)
+                for other_rank in range(_1d_mesh.size(dim=0)):
+                    if tp_rank != other_rank:
+                        slice_idx = [
+                            slice(other_rank * local_shape[0], (other_rank + 1) * local_shape[0]),
+                            slice(local_shape[1]),
+                        ]
+                        if enable_distribute_flag:
+                            # each rank within a TP group shall initialize local weights differently
+                            self.assertNotEqual(tensor_gather[slice_idx], tensor_local)
+                        else:
+                            # without the parallel RNG, weight initialization violates the TP setup:
+                            # each rank within a TP group has the same initial weights
+                            self.assertEqual(tensor_gather[slice_idx], tensor_local)
+
+                # check across TP groups
+                # all-gather local shards
+                tensor_gather = device_mesh.all_gather(tensor_local, mesh_dim=1, gather_dim=0)
+                for other_rank in range(device_mesh.size(dim=1)):
+                    if dp_rank != other_rank:
+                        slice_idx = [
+                            slice(other_rank * local_shape[0], (other_rank + 1) * local_shape[0]),
+                            slice(local_shape[1]),
+                        ]
+                        if enable_distribute_flag:
+                            # local weights shall be initialized the same acorss TP groups
+                            self.assertEqual(tensor_gather[slice_idx], tensor_local)
+                        else:
+                            # without the parallel RNG, weight initialization violates the TP setup:
+                            # local weights are initialized differently acorss TP groups due to different
+                            # random seeds set in data loading.
+                            self.assertNotEqual(tensor_gather[slice_idx], tensor_local)
+
+    @with_comms
+    @skip_if_lt_x_gpu(4)
+    def test_dropout_layer(self):
+        local_shape = [16, 6]
+
+        # in self-attention layer, the activation needs to go through 2 dropout layers:
+        #   1. attention dropout which belongs to a tensor-parallel region
+        #   2. residual dropout which belongs to a non-tensor-parallel region
+        mesh = torch.arange(self.world_size).reshape(2, 2)
+        device_mesh = DeviceMesh(self.device_type, mesh)
+        tp_rank = device_mesh.get_coordinate()[0]  # the tensor parallel dimension is 0
```

Comment:
[parallelize_module](https://github.com/pytorch/pytorch/blob/main/torch/distributed/tensor/parallel/api.py#L35-L40) has the default `tp_mesh_dim` value 0. I tried printing the 1-d mesh created and the result is:
device_mesh: [[0, 1], [2, 3]]
1d_mesh created in TP api: [0, 2] and [1, 3]

Reply:
Oh I think for the case of 2-D parallel what we are doing now is mainly tp_dim = 1, so user might need to specify their tp_dim when combining with data parallel. I think the default argument specified as 0 because there might be case we only want to perform tensor parallel without data parallel, in this case it's a 1-D mesh.

But I guess for the case of checking the content whether tp is dim 0 or dim 1 does not matter that much

Reply:
I  agree. For unit test this shouldn't bother.